homeassistant/components/ozw/services.py

```diff
@@ -126,15 +91,51 @@ def async_set_config_parameter(self, service):
                     return
                 payload = int(selection)
 
-            value.send_value(payload)  # send the payload
-            _LOGGER.info(
-                "Setting configuration parameter %s on Node %s with value %s",
-                param,
-                node_id,
-                payload,
-            )
+            # iterate list labels to get value
+            for selected in value.value["List"]:
+                if selected["Label"] != selection:
+                    continue
+                payload = int(selected["Value"])
```

Comment:
Is it correct that we want to prefer this value over the selection value as payload?

Reply:
Resolved in latest

Reply:
When does it happen that the loop here never sets the payload and the payload set on line 92 remains set?

Reply:
If I insert a `if isinstance(selection, string):` before the for loop that'd be a good guard then?

Reply:
What do we want to happen? I don't know what the goal is.

Reply:
The goal is to find the string label in the list and send the int of that value to MQTT, and allow lists to be set via a string or an int.

Reply:
We're only iterating the list in case the selection is a string. That doesn't look correct.

Reply:
Should be sorted in the latest commit.

Reply:
Why do we only want to iterate the list if the selection is not an integer?

Reply:
I think I see where you're going with this now. I'll see what I can do.

Reply:
There we go, now it's really sorted in the latest commit.